apps/condo/domains/user/integration/sberid/SberIdIdentityIntegration.js

```diff
@@ -0,0 +1,163 @@
+
+const { normalizeEmail } = require('@condo/domains/common/utils/mail')
+const { normalizePhone } = require('@condo/domains/common/utils/phone')
+
+// get sber id configuration params
+const SBER_ID_CONFIG = process.env.SBER_ID_CONFIG ? JSON.parse(process.env.SBER_ID_CONFIG) : {}
+const {
+    authorizeUrl,
+    tokenUrl,
+    userInfoUrl,
+    scope,
+    clientId,
+    clientSecret,
+    redirectUri,
+    cert,
+    key,
+    verifyServerSsl,
+} = SBER_ID_CONFIG
+const callbackPath = '/api/sber_id/auth/callback'
+const callbackUri = redirectUri || `${conf.SERVER_URL}${callbackPath}`
+const axiosTimeout = 10000
+
+// instantiate httpsAgent in order to support mTLS communication with sber id servers
+const httpsAgent = new https.Agent({
+    rejectUnauthorized: verifyServerSsl,
+    cert,
+    key,
+})
+
+// instantiate request id generator
+const nanoid = customAlphabet('1234567890abcdef', 32)
+
+class SberIdIdentityIntegration {
+    async generateLoginFormParams (checks, redirectUrl) {
+        const { nonce, state } = checks
+        const link = new URL(authorizeUrl)
+
+        // generate params
+        const responseType = 'code'
+        const redirectUri = isNil(redirectUrl) ? callbackUri : redirectUrl
+
+        // set params to link
+        link.searchParams.set('scope', scope)
+        link.searchParams.set('state', state)
+        link.searchParams.set('response_type', responseType)
+        link.searchParams.set('client_id', clientId)
+        link.searchParams.set('redirect_uri', redirectUri)
+        link.searchParams.set('nonce', nonce)
+
+        return link
+    }
+
+    async issueExternalIdentityToken (code, redirectUrl) {
+        const redirectUri = isNil(redirectUrl) ? callbackUri : redirectUrl
+        // set issue token request parameters
+        const request = {
+            grant_type: 'authorization_code',
+            code,
+            redirect_uri: redirectUri,
+            client_id: clientId,
+            client_secret: clientSecret,
+        }
+
+        // send a request
+        const tokenResponse = await axios.create({
+            timeout: axiosTimeout,
+            headers: {
+                'x-ibm-client-id': clientId,
+                'rquid': nanoid(),
+            },
+            validateStatus: () => true,
+            httpsAgent,
+        }).post(tokenUrl, new URLSearchParams(request))
+
+        // extract required params
+        const {
+            access_token: accessToken,
+            token_type: tokenType,
+            expires_in: expiresIn,
+            scope,
+            id_token: idToken,
+        } = tokenResponse.data
+
+        if (tokenResponse.status !== 200 || isNil(accessToken) || isNil(idToken)) {
+            throw new Error(JSON.stringify(tokenResponse.data))
+        }
+
+        return {
+            accessToken,
+            tokenType,
+            expiresIn,
+            scope,
+            idToken,
+        }
+    }
+
+    async getUserInfo ({ accessToken }) {
```

Comment:
Please, add some guards

```suggestion
    async getUserInfo ({ accessToken }) {
        if (!accessToken) throw new Error('call getUserInfo without required accessToken')
```

Reply:
done